helpers/services.js

```diff
@@ -52,3 +59,34 @@ export function sortByAddress(services, selectedAddress) {
   )
   return [serviceWithAddress, ...servicesWithoutSpecified]
 }
+
+export const isExtension = service =>
+  service?.method === SERVICE_METHODS.EXTENSION
+
+export const isExtensionInstalled = (extensions, address) =>
+  extensions.some(extension => extension?.provider?.address === address)
+
+// Filter out extensions in service list if they are installed
+export function filterServicesForInstalledExtensions(extensions = []) {
+  return function (services = []) {
+    return services.filter(
+      service => !isExtensionInstalled(extensions, service?.provider?.address)
+    )
+  }
+}
+
+export const requiresPlatform = service => {
+  const requiredPlatformTypes = [SERVICE_METHODS.EXTENSION]
```

Comment:
Should this be a flag on the flag to prevent issues with more wallet types being added in the future?